### DISCUSSION TITLE
updateByQuery ignore query

### DISCUSSION THREAD

**Discussion by gkielwasser** [directus/directus#9634] [Archive]
Hi,

Before opening an issue I would like to double check it is correct to update this way. 
It seems broken since I have updated from rc.98 to to v9.0.0

```
await userService.updateByQuery({ email: 'some@email'}, {
   first_name,
   last_name,
})
```

This is updating all user entries instead of only the one matching the given email.

Thanks

**Comment by azrikahar** [comment]
Hi! The query in question, should be an object containing any of the query parameters: https://docs.directus.io/reference/query/

So I think you are looking for the following code instead:

```js
await userService.updateByQuery({ filter: { email: 'some@email' } }, { first_name, last_name, })
```

> It seems broken since I have updated from rc.98 to to v9.0.0

As for this statement, I believe it has been _requiring_ query parameters as the keys ever since `updateByQuery` was first created 🤔 Perhaps there's a mix up with `updateMany` here.

**Reply by gkielwasser** [reply]
Thanks a lot. I will not dig in too much as I was sure when this regression happened in my app.

Anyway the expected type for query expects this filter property.

Fixed! Thanks!
